src/Storage/Redux/Slices/Discovery.ts

```diff
@@ -99,6 +161,9 @@ export const DiscoverySlice = createSlice({
                 state.custom.push(bookmark)
             } else {
                 state.favorites.push(bookmark)
+                if (!state.favoriteRefs) state.favoriteRefs = []
+                const order = state.favoriteRefs.length
+                state.favoriteRefs.push(createDAppReference(bookmark, order))
             }
```

Comment:
Based on the comment above, we may need to fix this aswell

Reply:
Refactored as discussed to a hook based approach.